### PR TITLE
Added 'how to use' to Buyer's Guide's product page

### LINKED ISSUE #2294: Add 'How to use this guide' link
Users are asking for more information regarding our research (why our categories are important, why they should care, etc) and the 'How to use this guide' content answers those questions. Since it's too much to fit into the product page itself without making the page too busy we're going to include a link to the About page that already has the 'How to use this guide' content.

- [x] Add 'Our Research' title

- [x] Add 'How to use this guide' link, and link to About page

- [x] Hover state #3302FF

**Assets:**
Grey information button: [grey-information-icon.svg.zip](https://github.com/mozilla/foundation.mozilla.org/files/2625912/grey-information-icon.svg.zip)

test b screen 1 copy 11 2x

### PR DESCRIPTION
**Pull request by mmmavis**
Closes #2294

### CODE REVIEW

File: source/sass/buyers-guide/views/product.scss [hunk 1]
```diff
@@ -318,6 +318,10 @@ $product-detail-padding-x: (
     width: 16px;
     height: 100%;
```
[the remark this answers is not in the document]
Comment (mmmavis) [reply source/sass/buyers-guide/views/product.scss]: We might not need this. Let me check with @beccaklam 

@beccaklam how do you want that link to wrap on small screen? This is what it looks like currently.

[the remark this answers is not in the document]
Comment (mmmavis) [reply source/sass/buyers-guide/views/product.scss]: I think the link will fit nicer if we move it to the next line. What do you think?

[the remark this answers is not in the document]
Comment (beccaklam) [reply source/sass/buyers-guide/views/product.scss]: > I think the link will fit nicer if we move it to the next line. What do you think?

@mmmavis Agreed!

**Comment by mmmavis** [comment]
PR updated! Made `How to use this guide` sit on its own line on [extra small](https://getbootstrap.com/docs/4.1/layout/grid/#grid-options) screens.

**Comment by mmmavis** [comment]
@Pomax this is ready for code review again & icon position should look better now

**Comment by mmmavis** [comment]
ping @Pomax 😉

**Comment by Pomax** [comment]
blerf! sorry for the delay, this looks great!

**Pull request merged**